eclair-core/src/main/scala/fr/acinq/eclair/db/PaymentsDb.scala

```diff
@@ -19,70 +19,131 @@ package fr.acinq.eclair.db
 import java.util.UUID
 
 import fr.acinq.bitcoin.ByteVector32
-import fr.acinq.eclair.MilliSatoshi
-import fr.acinq.eclair.payment.PaymentRequest
+import fr.acinq.bitcoin.Crypto.PublicKey
+import fr.acinq.eclair.payment._
+import fr.acinq.eclair.router.Hop
+import fr.acinq.eclair.{MilliSatoshi, ShortChannelId}
 
 trait PaymentsDb {
 
-  // creates a record for a non yet finalized outgoing payment
+  /** Create a record for a non yet finalized outgoing payment. */
   def addOutgoingPayment(outgoingPayment: OutgoingPayment)
 
-  // updates the status of the payment, if the newStatus is SUCCEEDED you must supply a preimage
-  def updateOutgoingPayment(id: UUID, newStatus: OutgoingPaymentStatus.Value, preimage: Option[ByteVector32] = None)
+  /** Update the status of the payment in case of success. */
+  def updateOutgoingPayment(paymentResult: PaymentSent)
 
+  /** Update the status of the payment in case of failure. */
+  def updateOutgoingPayment(paymentResult: PaymentFailed)
+
+  /** Get an outgoing payment attempt. */
   def getOutgoingPayment(id: UUID): Option[OutgoingPayment]
 
-  // all the outgoing payment (attempts) to pay the given paymentHash
+  /** Get all the outgoing payment attempts to pay the given paymentHash. */
   def getOutgoingPayments(paymentHash: ByteVector32): Seq[OutgoingPayment]
 
-  def listOutgoingPayments(): Seq[OutgoingPayment]
+  /** Get all the outgoing payment attempts in the given time range. */
+  def listOutgoingPayments(from: Long, to: Long): Seq[OutgoingPayment]
 
+  /** Add a new payment request (to receive a payment). */
   def addPaymentRequest(pr: PaymentRequest, preimage: ByteVector32)
```

Comment:
One could say that calling an invoice an "incoming payment" is a bit premature. Who knows if there will indeed be an incoming payment for this invoice... But `addPaymentRequest` is indeed too ambiguous.

Why not `addIncomingPayment`/`setPaidIncomingPayment` to be consistent with your suggestions below?

Reply:
> [...] calling an invoice an "incoming payment" is a bit premature. Who knows if there will indeed be an incoming payment for this invoice[...]

A node generates an invoice only because it expects this invoice to be paid with an incoming payment. Of course this payment may never actually come, just as an outgoing payment may never be actually sent, though we will store it all the same.

`setPaidIncomingPayment ` sounds great!